### DISCUSSION TITLE
Centering image from Image component

### DISCUSSION THREAD

**Discussion by brunonardini** [vercel/next.js#18388] [Help] [answered]
I switched my `<img />` element to the new Next.js Image component, and tried to center the image like this:

```jsx
return (
    <>
       {/* ..... */}
            <div className="image-column">
              <Image
                src="/images/logo_200px_transp.png"
                alt="Nardini Academy logo"
                width={200}
                height={200}
              />
            </div>
       {/* ..... */}
           
      <style jsx>{`
          .image-column {
            text-align: center;
          }
      `}</style>
    </>
  );
```

The resulting HTML was rendered as follows:

```html
<div class="jsx-39787449 image-column">
  <div style="max-width: 100%; width: 200px;">
    <div style="position: relative; padding-bottom: 100%;">
      <img
        alt="Nardini Academy logo"
        data-src="/_next/image?url=%2Fimages%2Flogo_200px_transp.png&amp;w=320&amp;q=75"
        data-srcset="/_next/image?url=%2Fimages%2Flogo_200px_transp.png&amp;w=320&amp;q=75 320w"
        class=""
        style="
          visibility: visible;
          height: 100%;
          left: 0px;
          position: absolute;
          top: 0px;
          width: 100%;
        "
        src="/_next/image?url=%2Fimages%2Flogo_200px_transp.png&amp;w=320&amp;q=75"
        srcset="
          /_next/image?url=%2Fimages%2Flogo_200px_transp.png&amp;w=320&amp;q=75 320w
        "
      />
    </div>
  </div>
</div>
```

I tried in several ways to center the image and the only way I got was to go straight to the element that looks like this:

```html
<div style="max-width:100%;width:200px;">
```

And exchange it for:

```html
<div style="max-width:100%;width:200px;display: inline-block;">
```

But I couldn't do that from the code. 

<table>
<tr>
<th>Original code</th>
<th>With the adjustment (expected result)</th>
</tr>
<tr>
<td>
<img src="https://user-images.githubusercontent.com/673904/97441727-c970a400-1907-11eb-9a53-d6c1511a8de5.png">
</td>
<td>
<img src="https://user-images.githubusercontent.com/673904/97441742-cf668500-1907-11eb-8597-e7804b68351d.png">
</td>
</tr>
</table>

Can you help me with this?

**Comment by jamesmosier** [comment, marked as answer]
This should work 😄 

```css
.image-column > div {
  margin: 0 auto;
}
```

**Reply by XavierGeerinck** [reply]
How would you recommend to fix this with CSS frameworks such as Tailwind? This feels hacky :)

**Reply by g-elwell** [reply]
Unfortunately because of how the `Image` component is working you are unable to style it's wrapping `div` directly. Any styles or classnames you add to the component get applied to the `img` element and not the wrapping `div` 😕

Since there's no way to add a classname to that wrapper, you can't style it with Tailwind.

As a result, adding your own wrapper (which you can control) and using CSS to select the immediate child element is the only option.

**Reply by XavierGeerinck** [reply]
Indeed, would love to see a className for the wrapper specifically! How I solve this personally in other components is by creating a classNameWrapper property actually. Maybe something like that could be added?

**Reply by g-elwell** [reply]
I actually think that classes/styles added to the `Image` component should apply to the wrapper by default, since the most of the time you'll be adding margin etc and you could still style the `img` directly via CSS if needed eg `.my-image img { ... }`. I've just opened a feature request to suggest this here #18585

**Reply by brunonardini** [reply]
@jamesmosier that code didn't work :cry: 

I had to add an extra div to be able to control it in the CSS:

```jsx
<div className="image-column">
    <div className="image-container">
        <Image
            src="/images/logo_200px_transp.png"
            alt="Nardini Academy logo"
            width={200}
            height={200}
        />
    </div>
</div>
```
Then in CSS I put the div as inline-block

```css
.image-column {
  text-align: center;
}

.image-container {
  display: inline-block;
}
```

**Reply by XavierGeerinck** [reply]
@g-elwell thanks for the issue! Definitely helpful 😊 doesn't matter how it's implemented for me, as long as it works without the hacks around it 😀

**Reply by brunonardini** [reply]
@g-elwell It could give us the option to change the CSS of both, the wrapper divs and the img itself.

**Reply by hugopiso** [reply]
I've got the exact same issue, and Im using **tailwind** so it's quite a dirty fix for me to create a wrapper but no choices..

Here is what I did:
``` html
 <div className="img-center">
        <Image src="/images/logo_200px_transp.png" width={200} height={200} />
 </div>
```

``` css
.img-center > div {
    display: block !important;
    margin: auto !important;
    max-width: max-content !important;;
}
```
